pkg/agent/agent.go

```diff
@@ -122,21 +124,14 @@ func (a *Agent) Shutdown() error {
 	if a.Catalog != nil {
 		a.Catalog.Stop()
 	}
-
+	a.stop <- struct{}{}
 	a.grpcServer.GracefulStop()
 
 	// Drain error channel, last one wins
 	var err error
-Drain:
-	for {
-		select {
-		case e := <-a.config.ErrorCh:
-			err = e
-		default:
-			break Drain
-		}
+	for err = range a.config.ErrorCh {
+		a.config.Log.Errorf(err.Error())
```

Comment:
When ranging over a channel, the channel must be explicitly closed in order to break the range - otherwise, it will block forever